hexa/pipelines/migrations/0046_pipelineversion_unique_pipeline_version_name.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import re
+
+from django.db import migrations, models
+
+logger = logging.getLogger(__name__)
+
+
+def add_timestamp_to_duplicates(apps, schema_editor):
+    PipelineVersion = apps.get_model("pipelines", "PipelineVersion")
+    duplicates = (
+        PipelineVersion.objects.values("pipeline", "name")
+        .annotate(count=models.Count("id"))
+        .filter(count__gt=1)
+    )
+
+    for duplicate in duplicates:
+        versions = PipelineVersion.objects.filter(
+            pipeline=duplicate["pipeline"], name__contains=duplicate["name"]
+        ).order_by("created_at")
+
+        for index, version in enumerate(
+            list(versions)[:-1]
+        ):  # Skip the most recent version, keep it unchanged
+            old_name = version.name
+            version.name = f"{old_name} (v{index+1})"
```

Comment:
Is this acceptable for our users ?

Reply:
Yes. I'm not sure about the transformation though. You leave the latest one untouched ("My Version") and then you take all the older version ordered by creation_date to have in the end:
"My Version (1)"; "My Version (2)"; "My Version (3)"; "My Version". 

It's a bit counterintuitive since it does not follow what users are used to with their FS. I would apply the same change to the latest one as well. WDYT?

Reply:
My reasoning is that the last one is the most important because it's the one used. So I wanted to keep it untouched so that it's not a breaking change
WDYT ?